### DISCUSSION TITLE
Documents upload is taking forever

### DISCUSSION THREAD

**Discussion by kiranch97** [open-webui/open-webui#5968] [General]
# Bug Report

**Important: Before submitting a bug report, please check whether a similar issue or feature request has already been posted in the Issues or Discussions section. It's likely we're already tracking it. In case of uncertainty, initiate a discussion post first. This helps us all to efficiently focus on improving the project.**

**Let's collaborate respectfully. If you bring negativity, please understand our capacity to engage may be limited. If you're open to learning and communicating constructively, we're more than happy to assist you. Remember, Open WebUI is a volunteer-driven project maintained by a single maintainer, supported by our amazing contributors who also manage full-time jobs. We respect your time; please respect ours. If you have an issue, We highly encourage you to submit a pull request or to fork the project. We actively work to prevent contributor burnout to preserve the quality and continuity of Open WebUI.**

## Installation Method

1. we have deployed Ollama and openwebui on separate pods and Ollama running on A100 GPU.

## Environment

- **Open WebUI Version:** v0.3.30
- **Ollama (if applicable):** 0.3.12

- **Operating System:**  Windows 11
- **Browser (if applicable):** Chrome Version 129.0.6668.59 (Official Build) (64-bit)

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
[docker_logs.txt](https://github.com/user-attachments/files/17274138/docker_logs.txt)

[Describe what you expected to happen.]

## Actual Behavior:

[Describe what actually happened.]

## Description

**Bug Summary:**
, issue is when the user upload the document, for few MB size as well its taking forever to upload state, this is on internal network, what I understand the vector data store is not process fast enough, can you please advise whether any fine-tune to be done?

and also one more observation is that, when document uploading state, the CPU on openweb-ui pod hits to the max limit, if we increase the pod cpu to 30Cores as well it spike to 30core usage. 

## Reproduction Details

**Steps to Reproduce:**
anytime issue can be re-produce, by simply uploading the any document. 

## Logs and Screenshots
[docker_logs.txt](https://github.com/user-attachments/files/17274389/docker_logs.txt)
![uploading state](https://github.com/user-attachments/assets/ff157b93-61be-4242-a0a4-5b6ac307999c)

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots/Screen Recordings (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by kiranch97** [comment]
[10.8.64.163.zip](https://github.com/user-attachments/files/17274403/10.8.64.163.zip)
browser log file

**Comment by kiranch97** [comment]
![pods metrics](https://github.com/user-attachments/assets/4d53b64d-421d-486f-9bfd-2b56dcda8abd)
Pod metrics

**Comment by thiswillbeyourgithub** [comment]
When sending a file it is chunked and embeddings are created. I think that's what's happening. If you have limited power or want faster processing use external embedding api?

**Reply by kiranch97** [reply]
Hi @thiswillbeyourgithub Thanks, do you have available deployment model for external embedded api? please advise.

**Reply by thiswillbeyourgithub** [reply]
Jina api for example?

**Comment by kiranch97** [comment]
Hi @thiswillbeyourgithub 

can I use: https://huggingface.co/jinaai/jina-embeddings-v3

**Comment by rodrigo-martins-multiedro** [comment]
Hello, you are having problems uploading documents in the chat to ask specific questions, I noticed that files up to 900kb it processes without problems, but when it goes beyond that it keeps loading forever, there have been cases where it uploads but when I ask in the model there is no reference to the file, I am reading this post and I was wondering if there was a solution and if so what it was. I'm testing on two versions and the behavior is always the same v0.5.9 
v0.3.16

I have already removed the limit on size and number of files, but without success.

I am using GPT models during chat interaction

**Comment by dokalanyi** [comment]
I have the same issue, on a 3090 and i7. And it's slow for small files as well, the ECU and GPU stay idle

I tried webui  and same thing happened but I'm now using the API:

curl -X POST -H "Authorization: Bearer XXX" -H "Accept: application/json" \
-F "file=@/Users/xxx/file.txt" http://192.168.1.10:3000/api/v1/files/

**Reply by rodrigo-martins-multiedro** [reply]
I discovered some important things in the environment I'm setting up, in my case here I'm running openwebui on doker and I have nginx between the internet and the app, in nginx I had to make adjustments to the following parameters and now I can upload the files, the upload is still very slow but at least now I can upload files to carry out the processing, below are the parameters that I adjusted in the /etc/nginx/sites-available/ file relating to my website and I adjusted the following parameters 

```
     # SSL Security
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers HIGH:!aNULL:!MD5;
    ssl_ecdh_curve secp384r1:X25519;
    ssl_prefer_server_ciphers on;
```

I had to make this adjustment because I was noticing some ssl errors during requests 

here is the configuration that really had a result here we are adjusting the limit for uploading the file in the variable 
`        client_max_body_size 100M;`

Another extremely important point is the timeout of the sessions. I realized that as the upload process took a few moments to end, I had to adjust the parameters below so that the session would not be terminated. 
```
        proxy_read_timeout 1800s;
        proxy_connect_timeout 1800s;
        proxy_send_timeout 1800s;
```

Here is the complete configuration of my website file

```
server {
    listen 80;
    server_name <your site .com>;
    return 301 https://$host$request_uri;
}

server {
    listen 443 ssl;
    server_name <your site .com>;

    # Certificados SSL (ajuste os caminhos se necessário)
    ssl_certificate /app/ssl/<certificate your>.crt;
    ssl_certificate_key /app/ssl/<certificate your>.key;
    ssl_trusted_certificate /app/ssl/intermediario.crt;

    # Segurança SSL
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers HIGH:!aNULL:!MD5;
    ssl_ecdh_curve secp384r1:X25519;
    ssl_prefer_server_ciphers on;

    location / {
        proxy_pass http://localhost:8080;  # Ajuste para a porta do OpenWebUI
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        # Permitir uploads grandes e evitar timeouts
        client_max_body_size 100M;
        proxy_read_timeout 1800s;
        proxy_connect_timeout 1800s;
        proxy_send_timeout 1800s;

        # Melhorar cache e performance do proxy
        proxy_buffering on;
        proxy_buffers 8 16k;
        proxy_buffer_size 32k;
        proxy_busy_buffers_size 64k;
        proxy_temp_file_write_size 64k;
    }

    # Página de erro customizada para 50x
    error_page 500 502 503 504 /50x.html;
    location = /50x.html {
        root /usr/share/nginx/html;
    }
}
```

**Reply by rodrigo-martins-multiedro** [reply]
After these adjustments I was able to upload the files and ask the questions with the desired model but the upload is very time consuming, even a 1mb file takes a long time and if it is a larger file it will take longer, I don't know if it is some other configuration in my environment or in the openweb UI itself but I managed to advance a little further.

**Reply by dokalanyi** [reply]
I'm not using nginx, so connecting direct to the docker, and smaller files take long, however anything bigger than 1MB takes forever.

**Comment by an0mal1a** [comment]
Still this problem, and im using Open-WebUI without docker cause it take so long to show the response from the model (ollama). Now without docker responses are faster but file upload are very very slow...

**Reply by rgaricano** [reply]
shouldn't, do you have  env var USE_CUDA_DOCKER="true" setted? (set it even if you don't use docker)

**Reply by an0mal1a** [reply]
Hey, I have an older PC with modest specs and a low-end GPU that doesn’t support CUDA. After installing all the drivers, I realized this GPU can’t accelerate AI workloads via CUDA. The main purpose of this server is to run AI models, but **I managed to optimize the upload** settings in the administrator config.

My current issue is with RAG and processing large CSV files. I set up Apache Tika (both /rmeta/text and /tika endpoints has the same results) for extraction, but when I query specific IDs or text from the CSV, the extraction doesn’t work well. It seems to be related to the extraction step.

I tested different extraction models, such as:

- bge-reranker-v2-m3
- bge-small-en-v1
- etc

But the results are the same. Also, loading and processing the full context is very slow.

The specs are:

- CPU: Ryzen 5 3600
- RAM: 32GB DDR4 2400MHz
- Motherboard: Asus B450-e Gaming

Is there a known solution for this?

**Reply by rgaricano** [reply]
I use a similar pc, the hybrid search is not more usefull in this case, indeed rerank slowdown all the process (is one more model that have to load and run in cpu), you can supply hybrid search with a improved prompt.
No so much to do, I encountered the same problems (It also couldn't generate images in good conditions). My solution was change the GPU for a cheap, but usable, RTX3060 12GB and the improvement was remarkable. It is not what I would like but it works quite well for me to perform all kinds of tests, it runs models up to 14b, with limited contexts (8192) but more than enough for the use I give it.

Anyway, if you still use hybrid search, in your case I would adjust it to more lexical (Weight of BM25 Retrieval near 0) since in a cvs the semantics are not very relevant and what we need is accuracy in the term of search.

**Comment by RC-1290** [comment]
If this needs to take a long time, it would be nice if:
1. The upload could be completed first, so that the processing can happen without having the tab open.
2. There was a visualization of the progress (overall and individual files), with a way to cancel the upload.

At the moment you have to catch it finishing a file, at which point there is a notification showing the progress, but depending on the file size these messages can be quite a time apart. It is unclear what you can and can't do on the same tab while the upload is progressing (you can apparently click around the openwebui GUI without the upload stopping).

**Comment by Exotik850** [comment]
Has there any work been done on this? Uploading ~5K 10-100kb files into a DB and it's taking absolute ages to upload, would really like anything that can speed it up
